delorean/dates.py

```diff
@@ -175,51 +181,65 @@ def __init__(self, datetime=None, timezone=None):
         # use UTC
         is_datetime_instance(datetime)
 
-        naive = True
-        self._tz = timezone
-        self._dt = datetime
-
-        if not is_datetime_naive(datetime):
-            naive = False
-            self._tz = datetime.tzinfo.tzname(None)
-            self._dt = datetime
-
-        if naive:
-            if timezone is None and datetime is None:
-                self._tz = UTC
-                self._dt = datetime_timezone(UTC)
-            elif timezone is not None and datetime is None:
-                # create utctime then normalize to tz
-                self._tz = timezone
-                self._dt = datetime_timezone(timezone)
-            elif timezone is None and datetime is not None:
-                raise DeloreanInvalidTimezone('Provide a valid timezone')
+        if datetime:
+            if is_datetime_naive(datetime):
+                if timezone:
+                    if isinstance(timezone, tzoffset):
+                        self._tzinfo = pytz.FixedOffset(timezone.utcoffset(None).total_seconds() / 60)
+                    elif isinstance(timezone, tzinfo):
+                        self._tzinfo = timezone
+                    else:
+                        self._tzinfo = pytz.timezone(timezone)
+                    self._dt = localize(datetime, self._tzinfo)
+                    self._tzinfo = self._dt.tzinfo
+                else:
+                    #TODO(mlew, 2015-08-09):
```

Comment:
I could go either way on this for the time being, although I do think we ultimately should return a Delorean object in UTC. In my experience, working with datetimes is almost always in UTC, and it would be good to provide a little bit of magic for users here.  We will obviously need to change the documentation.

Part of the draw of using Delorean over datetime should be that it helps inexperienced developers learn how to properly work with datetime objects, and ALWAYS working with UTC unless specified otherwise is a great default to have for n00bs that don't know any better.